src/main/java/ui/UI.java

```diff
@@ -217,6 +221,16 @@ private void initPreApplicationState() {
 
         uiManager = new UIManager(this);
         status = new HTStatusBar(this);
+
+        updateProgressWindow = new UpdateProgressWindow();
+
+        if (!TestController.isTestMode()) {
+            updateManager = new UpdateManager(updateProgressWindow);
+        } else {
+            updateManager = new UpdateManagerStub();
```

Comment:
I tried mockito to mock it but currently mockito is only referenced in tests and not in the main application. I don't think introducing mocking in TestController would be a good idea.